### PR TITLE
Istio and istio-configuration components

### PR DESCRIPTION
**Pull request by werdes72**
See #698

### CODE REVIEW

**Review by tobiscr — DISMISSED** [review]

**Review by mjakobczyk — CHANGES_REQUESTED** [review]
Your solution does not work as expected. Please see my [comment](https://github.com/kyma-incubator/reconciler/issues/698#issuecomment-1026528213) for more details.

File: pkg/reconciler/instances/istio/action.go [hunk 1]
```diff
@@ -113,7 +112,83 @@ func (a *ReconcileAction) Run(context *service.ActionContext) error {
...
+	getIstioPerformer bootstrapIstioPerformer
+}
+
+// NewReconcileIstioConfigurationAction returns an instance of ReconcileIstioConfigurationAction
+func NewReconcileIstioConfigurationAction(getIstioPerformer bootstrapIstioPerformer) *ReconcileAction {
+	return (*ReconcileAction)(&ReconcileIstioConfigurationAction{getIstioPerformer})
+}
+
+func (a *ReconcileIstioConfigurationAction) Run(context *service.ActionContext) error {
+	context.Logger.Debugf("Reconcile action of istio-configuration triggered")
```
Comment (mjakobczyk) [inline pkg/reconciler/instances/istio/action.go:158]: Should it use formatted log when no formatting is used?

File: pkg/reconciler/instances/istio/action.go [hunk 2]
```diff
@@ -113,7 +112,83 @@ func (a *ReconcileAction) Run(context *service.ActionContext) error {
...
+
+	return nil
+}
+
+type ReconcileIstioConfigurationAction struct {
+	getIstioPerformer bootstrapIstioPerformer
+}
+
+// NewReconcileIstioConfigurationAction returns an instance of ReconcileIstioConfigurationAction
+func NewReconcileIstioConfigurationAction(getIstioPerformer bootstrapIstioPerformer) *ReconcileAction {
```
Comment (mjakobczyk) [inline pkg/reconciler/instances/istio/action.go:153]: Why do you return `ReconcileAction` which belongs to `istio` component?

File: pkg/reconciler/instances/istio/action.go [hunk 3]
```diff
@@ -113,7 +112,83 @@ func (a *ReconcileAction) Run(context *service.ActionContext) error {
...
+	return (*ReconcileAction)(&ReconcileIstioConfigurationAction{getIstioPerformer})
```
Comment (mjakobczyk) [inline pkg/reconciler/instances/istio/action.go:154]: As above.

**Review by mjakobczyk — CHANGES_REQUESTED** [review]

**Review by mjakobczyk — APPROVED** [review]
Much better after changes, LGTM 👍

**Pull request merged**